### PR TITLE
core: document conflict detection endpoint API

### PR DESCRIPTION
**Pull request by eckter**

### CODE REVIEW

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionRequest.kt [hunk 1]
```diff
@@ -11,20 +11,46 @@ import fr.sncf.osrd.utils.json.UnitAdapterFactory
 import java.time.ZonedDateTime
 
 class ConflictDetectionRequest(
+    /** Infra ID. */
     var infra: String,
+    /** Expected infra version. */
```
Comment (emersion) [inline core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionRequest.kt]: Maybe would be useful to add the purpose here. For instance, "Used for cache invalidation."?

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt [hunk 2]
```diff
@@ -9,22 +9,47 @@ import fr.sncf.osrd.utils.json.UnitAdapterFactory
 import java.time.ZonedDateTime
 
 class ConflictDetectionResponse(
+    /**
+     * List of all detected conflicts, if any. Conflicts are grouped between each set of conflicting
```
Comment (emersion) [inline core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: s/grouped between/grouped by/

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt [hunk 3]
```diff
@@ -9,22 +9,47 @@ import fr.sncf.osrd.utils.json.UnitAdapterFactory
...
+     * train/work schedule IDs. (i.e. there's only one conflict entry between trains (A, B), but
+     * there may be a different one for trains (A, B), or (B, C, D)).
```
Comment (emersion) [inline core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: This repeats (A, B) but I'm not sure this was intentional here?

**Review by emersion — APPROVED** [review]
Apart from these nits, LGTM!

This is consistent with the tests in `ConflictDetectionTest.testConflictMerge`.

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt [hunk 4]
```diff
@@ -9,22 +9,47 @@ import fr.sncf.osrd.utils.json.UnitAdapterFactory
...
+    /** One of "SPACING" or "ROUTING" depending on the kind of conflicting resource. */
     @Json(name = "conflict_type")
     val conflictType: ConflictDetectionEndpoint.ConflictDetectionResult.Conflict.ConflictType,
+    /** List of all conflicting requirements. Can't be empty. */
     @Json(name = "requirements") val requirements: Collection<ConflictRequirement>,
 )
 
+/**
+ * One conflicting requirement. The zone is used by at least two of the conflicting
+ * [trains / work schedule] in the given time range. This time range is the *union* and *not the
```
Comment (Erashin) [inline core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]:
```suggestion
 * [trains / work schedules] in the given time range. This time range is the *union* and *not the
```

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt [hunk 5]
```diff
@@ -9,22 +9,47 @@ import fr.sncf.osrd.utils.json.UnitAdapterFactory
...
+    /**
+     * List of all detected conflicts, if any. Conflicts are grouped between each set of conflicting
+     * train/work schedule IDs. (i.e. there's only one conflict entry between trains (A, B), but
+     * there may be a different one for trains (A, B), or (B, C, D)).
+     */
     val conflicts: Collection<Conflict>,
 )
 
+/**
+ * One conflict between a set of trains. If a given set of train is conflicting over a continuous
```
Comment (Erashin) [inline core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]:
```suggestion
 * One conflict between a set of trains and potentially a set of work schedules. If a given set of trains/work schedules is conflicting
```

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt [hunk 6]
```diff
@@ -9,22 +9,47 @@ import fr.sncf.osrd.utils.json.UnitAdapterFactory
...
     @Json(name = "conflict_type")
     val conflictType: ConflictDetectionEndpoint.ConflictDetectionResult.Conflict.ConflictType,
+    /** List of all conflicting requirements. Can't be empty. */
     @Json(name = "requirements") val requirements: Collection<ConflictRequirement>,
 )
 
+/**
+ * One conflicting requirement. The zone is used by at least two of the conflicting
+ * [trains / work schedule] in the given time range. This time range is the *union* and *not the
+ * intersection* of the time during which they are used by at least one train. i.e. if a train uses
```
Comment (Erashin) [inline core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: I do believe here, it's either a zone or a resource which is used.
```suggestion
 * intersection* of the time during which it is used by at least one train, i.e., if a train uses
```

**Review by Erashin — APPROVED** [review]
I'd like @Khoyo to check the core of this issue (pun intended :D) as he knows this part better than me. Other than the small following comments, this looks fine to me though :)

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionRequest.kt (hunk 1 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionRequest.kt]: Fixed in https://github.com/OpenRailAssociation/osrd/pull/9391/commits/649f0b097ad6f6412d4e5a8a1842faef22ee3e70

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt (hunk 2 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: Fixed in https://github.com/OpenRailAssociation/osrd/pull/9391/commits/649f0b097ad6f6412d4e5a8a1842faef22ee3e70

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt (hunk 3 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: Fixed in https://github.com/OpenRailAssociation/osrd/pull/9391/commits/649f0b097ad6f6412d4e5a8a1842faef22ee3e70

I did intend a kind of repetition but with the triplet ((A/B) and (A/B/C)), but that wasn't it

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt (hunk 4 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: Fixed in https://github.com/OpenRailAssociation/osrd/pull/9391/commits/649f0b097ad6f6412d4e5a8a1842faef22ee3e70

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt (hunk 5 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: Fixed in https://github.com/OpenRailAssociation/osrd/pull/9391/commits/649f0b097ad6f6412d4e5a8a1842faef22ee3e70

I didn't use the exact suggested phrasing though

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt (hunk 6 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: Fixed in https://github.com/OpenRailAssociation/osrd/pull/9391/commits/649f0b097ad6f6412d4e5a8a1842faef22ee3e70

File: core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt [hunk 7]
```diff
@@ -10,16 +10,17 @@ import java.time.ZonedDateTime
 
 class ConflictDetectionResponse(
     /**
-     * List of all detected conflicts, if any. Conflicts are grouped between each set of conflicting
+     * List of all detected conflicts, if any. Conflicts are grouped by each set of conflicting
```
Comment (Khoyo) [inline core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt]: by sets

**Review by Khoyo — APPROVED** [review]
Otherwise LGTM

**Review by Khoyo — APPROVED** [review]

**Pull request merged**